In [3]:
import numpy as np
import pandas as pd 
import matplotlib.pyplot as plt 
import seaborn as sns 


visa_df=pd.read_csv(r"C:\Users\Patel\Documents\NareshIt\EDA\Visadataset.csv")
visa_df


,case_id,continent,education_of_employee,has_job_experience,requires_job_training,no_of_employees,yr_of_estab,region_of_employment,prevailing_wage,unit_of_wage,full_time_position,case_status
0,EZYV01,Asia,High School,N,N,14513,2007,West,592.2029,Hour,Y,Denied
1,EZYV02,Asia,Master's,Y,N,2412,2002,Northeast,83425.6500,Year,Y,Certified
2,EZYV03,Asia,Bachelor's,N,Y,44444,2008,West,122996.8600,Year,Y,Denied
3,EZYV04,Asia,Bachelor's,N,N,98,1897,West,83434.0300,Year,Y,Denied
4,EZYV05,Africa,Master's,Y,N,1082,2005,South,149907.3900,Year,Y,Certified
...,...,...,...,...,...,...,...,...,...,...,...,...
25475,EZYV25476,Asia,Bachelor's,Y,Y,2601,2008,South,77092.5700,Year,Y,Certified
25476,EZYV25477,Asia,High School,Y,N,3274,2006,Northeast,279174.7900,Year,Y,Certified
25477,EZYV25478,Asia,Master's,Y,N,1121,1910,South,146298.8500,Year,N,Certified
25478,EZYV25479,Asia,Master's,Y,Y,1918,1887,West,86154.7700,Year,Y,Certified


In [6]:
num=visa_df.select_dtypes(exclude='object').columns
cat=visa_df.select_dtypes(include='object').columns
cat

Index(['case_id', 'continent', 'education_of_employee', 'has_job_experience',
       'requires_job_training', 'region_of_employment', 'unit_of_wage',
       'full_time_position', 'case_status'],
      dtype='object')

In [5]:
num

Index(['no_of_employees', 'yr_of_estab', 'prevailing_wage'], dtype='object')

**prevailing_wage**

In [3]:
visa_df['prevailing_wage']

0           592.2029
1         83425.6500
2        122996.8600
3         83434.0300
4        149907.3900
            ...     
25475     77092.5700
25476    279174.7900
25477    146298.8500
25478     86154.7700
25479     70876.9100
Name: prevailing_wage, Length: 25480, dtype: float64

- count

- min

- 25p

- 50p

- mean 

- median

- 75p

- max

In [5]:
visa_df['prevailing_wage']

0           592.2029
1         83425.6500
2        122996.8600
3         83434.0300
4        149907.3900
            ...     
25475     77092.5700
25476    279174.7900
25477    146298.8500
25478     86154.7700
25479     70876.9100
Name: prevailing_wage, Length: 25480, dtype: float64

In [10]:
pwage=visa_df['prevailing_wage']

In [8]:
nwqWQpwage=visa_df['prevailing_wage']
nwqWQpwage

0           592.2029
1         83425.6500
2        122996.8600
3         83434.0300
4        149907.3900
            ...     
25475     77092.5700
25476    279174.7900
25477    146298.8500
25478     86154.7700
25479     70876.9100
Name: prevailing_wage, Length: 25480, dtype: float64

In [9]:
pwage=visa_df['prevailing_wage']
Count=pwage.count()
Min= pwage.min()
P_25=round(np.percentile(pwage,25),2)
Mean=round(pwage.mean(),2)
Median=round(pwage.median(),2)
P_50=round(np.percentile(pwage,50),2)
P_75=round(np.percentile(pwage,75),2)
Max= pwage.max()
###############################################
l=[Count,Min,P_25,Mean,Median,P_50,P_75,Max]
####################################################
ID=['Count','Min','25P','Mean','Median','50P','75P','Max']
pd.DataFrame(l,columns=['prevailing_wage'],
            index=ID)

,prevailing_wage
Count,25480.0000
Min,2.1367
25P,34015.4800
Mean,74455.8100
Median,70308.2100
50P,70308.2100
75P,107735.5100
Max,319210.2700


In [10]:
# If i want to create a Df with 3 columns I need 3 lists
df=pd.DataFrame()
df['1']=[10,20]
df['2']=[20,30]
df

,1,2
0,10,20
1,20,30


In [15]:
final_list=[]
for i in num:
    Count=visa_df[i].count()
    Min= visa_df[i].min()
    P_25=round(np.percentile(visa_df[i],25),2)
    Mean=round(visa_df[i].mean(),2)
    Median=round(visa_df[i].median(),2)
    P_50=round(np.percentile(visa_df[i],50),2)
    P_75=round(np.percentile(visa_df[i],75),2)
    Max= visa_df[i].max()
    l=[Count,Min,P_25,Mean,Median,P_50,P_75,Max]
    final_list.append(l)

df=pd.DataFrame()
for i in final_list:
    df[f'{i}']=i
df.columns=num
df.index=ID
df

,no_of_employees,yr_of_estab,prevailing_wage
Count,25480.00,25480.00,25480.0000
Min,-26.00,1800.00,2.1367
25P,1022.00,1976.00,34015.4800
Mean,5667.04,1979.41,74455.8100
Median,2109.00,1997.00,70308.2100
50P,2109.00,1997.00,70308.2100
75P,3504.00,2005.00,107735.5100
Max,602069.00,2016.00,319210.2700


**Describe function**

In [17]:
visa_df.describe()

,no_of_employees,yr_of_estab,prevailing_wage
count,25480.000000,25480.000000,25480.000000
mean,5667.043210,1979.409929,74455.814592
std,22877.928848,42.366929,52815.942327
min,-26.000000,1800.000000,2.136700
25%,1022.000000,1976.000000,34015.480000
50%,2109.000000,1997.000000,70308.210000
75%,3504.000000,2005.000000,107735.512500
max,602069.000000,2016.000000,319210.270000


- data types seperate 

- columns seperated

- freqeuncy table

- math information


**Percentile**

In [19]:
wage_data=visa_df['prevailing_wage']
np.percentile(wage_data,25)

np.float64(34015.479999999996)

- 25percentage of number of applicants have the wages lower than **34015.479999999996**

- 25%(25480) has salary less than 34015.479999999996

- 6370 applicants has salary less than 34015.479999999996

In [20]:
wage_data=visa_df['prevailing_wage']
con=wage_data<np.percentile(wage_data,25)
np.sum(con)

np.int64(6370)

In [21]:
visa_df[con]

,case_id,continent,education_of_employee,has_job_experience,requires_job_training,no_of_employees,yr_of_estab,region_of_employment,prevailing_wage,unit_of_wage,full_time_position,case_status
0,EZYV01,Asia,High School,N,N,14513,2007,West,592.2029,Hour,Y,Denied
7,EZYV08,North America,Bachelor's,Y,N,3035,1924,West,418.2298,Hour,Y,Denied
12,EZYV13,Asia,Bachelor's,Y,N,123876,1963,Northeast,28663.0500,Year,Y,Certified
16,EZYV17,Europe,Master's,Y,N,76638,1991,Midwest,3706.7900,Year,Y,Certified
17,EZYV18,Asia,Master's,Y,N,2747,2001,West,16132.6100,Year,Y,Certified
...,...,...,...,...,...,...,...,...,...,...,...,...
25461,EZYV25462,Asia,Master's,Y,N,2861,2004,West,54.9196,Hour,Y,Denied
25465,EZYV25466,North America,High School,N,N,2577,1995,South,481.2238,Hour,Y,Certified
25466,EZYV25467,Europe,Bachelor's,Y,N,1938,2005,West,6973.4000,Year,Y,Denied
25470,EZYV25471,North America,Master's,Y,N,2272,1970,Northeast,516.4101,Hour,Y,Certified


**Emperical Rule**

 - 68% of data available between u-1*sd to u+1*sd

- step-1: calculate mean of wage: mean=74455

- step-2: calculate sd of wage: sd: 52815

- step-3: lb=mean-1*sd : 21640

- step-4: ub=mean+1*sd  : 127270

- step-5: con1= wage_data>lb

- step-6: con2= wage_data<ub

- step-7: con= con1 & con2

- step-8: np.sum(con): 17170

- step-9: 68%(25480) : 96328

In [22]:
# step-1: calculate mean of wage: mean=74455
wage_data=visa_df['prevailing_wage']
mean=wage_data.mean()
#step-2: calculate sd of wage: sd: 52815
sd=wage_data.std()
#step-3: lb=mean-1*sd : 21640
lb=mean-1*sd
#step-4: ub=mean+1*sd  : 127270
ub=mean+1*sd
#step-5: con1= wage_data>lb
con1= wage_data>lb
#step-6: con2= wage_data<ub
con2= wage_data<ub
# step-7: con= con1 & con2
con= con1 & con2
# step-8: np.sum(con): 17170
np.sum(con)==68*(25480)/100


np.False_

In [ ]:
wage_data=visa_df['prevailing_wage']
mean=wage_data.mean()
sd=wage_data.std()
lb=mean-1*sd
ub=mean+1*sd
con1= wage_data>lb
con2= wage_data<ub
con= con1 & con2
np.sum(con),68*(25480)/100
# Go to lab now onwards == real students 

In [16]:
Min=pwage.min(),
Max=pwage.max(),
P_25=np.percentile(pwage,25),
Mean=pwage.mean()
Median=pwage.median()

In [17]:
visa_df.describe()

,no_of_employees,yr_of_estab,prevailing_wage
count,25480.000000,25480.000000,25480.000000
mean,5667.043210,1979.409929,74455.814592
std,22877.928848,42.366929,52815.942327
min,-26.000000,1800.000000,2.136700
25%,1022.000000,1976.000000,34015.480000
50%,2109.000000,1997.000000,70308.210000
75%,3504.000000,2005.000000,107735.512500
max,602069.000000,2016.000000,319210.270000


**Describe function**

- data types seperate 

- columns seperated 

- frequency table

- math information`

**Percentile**

In [18]:
wage_data=visa_df['prevailing_wage']
np.percentile(wage_data,25)

np.float64(34015.479999999996)

- 25percentage of number of applicants havee the wages lower than **34015.479999999996**

- 25%(25480) has salary less than 34015.479999999996

- 6370 applicants has salary less than 34015.479999999996

In [27]:
visa_df['prevailing_wage']
con=wage_data<np.percentile(wage_data,25)
print(np.sum(con))

6370


In [28]:
visa_df[con]

,case_id,continent,education_of_employee,has_job_experience,requires_job_training,no_of_employees,yr_of_estab,region_of_employment,prevailing_wage,unit_of_wage,full_time_position,case_status
0,EZYV01,Asia,High School,N,N,14513,2007,West,592.2029,Hour,Y,Denied
7,EZYV08,North America,Bachelor's,Y,N,3035,1924,West,418.2298,Hour,Y,Denied
12,EZYV13,Asia,Bachelor's,Y,N,123876,1963,Northeast,28663.0500,Year,Y,Certified
16,EZYV17,Europe,Master's,Y,N,76638,1991,Midwest,3706.7900,Year,Y,Certified
17,EZYV18,Asia,Master's,Y,N,2747,2001,West,16132.6100,Year,Y,Certified
...,...,...,...,...,...,...,...,...,...,...,...,...
25461,EZYV25462,Asia,Master's,Y,N,2861,2004,West,54.9196,Hour,Y,Denied
25465,EZYV25466,North America,High School,N,N,2577,1995,South,481.2238,Hour,Y,Certified
25466,EZYV25467,Europe,Bachelor's,Y,N,1938,2005,West,6973.4000,Year,Y,Denied
25470,EZYV25471,North America,Master's,Y,N,2272,1970,Northeast,516.4101,Hour,Y,Certified


**Emperical Rule**

In [ ]:
- 68% of data available between u-1*sd to u+1*sd

- step-1:calculate mean of wage :
- step-2: calculate sd of wage :
- step-3: lowerbound=mean-1*sd:
- step-4: upperbound=mean+1*sd:
- step-5: con1=wage_data>lb
- step-6: con2= wage_data<ub
- step-7: con=con1 and con2 
- step-8: np.sum(con):
- step-9: 68%(25480)

In [29]:
wage_data=visa_df['prevailing_wage']

In [ ]:
**Conclusion is wage data does not follow normal  

**Histogram**

In [1]:
wage_data=visa_df['prevailing_wage']
plt.hist

NameError: name 'visa_df' is not defined